### DISCUSSION TITLE
Is it possible to add a command to fold/unfold the property section?

### DISCUSSION THREAD

**Discussion by XMB-7** [chrisgurney/obsidian-note-toolbar#89] [Ideas]
Or add an option in config to switch the behavior of the provided command `Toggle Properties` from show/hide the property section to fold/unfold the property section. 

I've been looking for a plugin that can fold/unfold the property section for a long time.

**Comment by chrisgurney** [comment]
@XMB-7 Thanks for the idea! 

Can you explain why you prefer to just fold the properties, versus completely hiding them (with the command in the plugin)?

**Comment by XMB-7** [comment]
Well, I think that directly hiding the YAML section is a more advanced option than just folding it, but it may cause users to forget its existence, especially for new users who might not have a clear conception of metadata in obsidian. If they accidentally trigger the command to hide the YAML section, or just forget to restore it after trying the command, they might forget about it over time. They may then start to wonder why their documents aren't displaying the YAML section and mistake it as a bug.
In comparison, just folding the YAML section at least reminds users that something exists there and even encourages users to fully utilize the power of document metadata.

**Reply by chrisgurney** [reply]
Appreciate the explanation!

Can't you just use the built-in `Toggle fold properties in current file` command? It sounds like that does what you want.

**Reply by XMB-7** [reply]
Wow, I have no idea there exists such a command. I've been always using `shell commands` to run a combination of several command URIs to fold the YAML section. Thank you let me know this.

**Reply by chrisgurney** [reply]
No problem! I totally forgot about that, but remembered I had looked into it when I was implementing the hide properties command. Glad that helps!

**Reply by XMB-7** [reply]
I just found that I still expect a command to exactly fold but not toggle the properties.
Here is my use case: I'd like to run a command URI via `shell commands` to automatically fold the properties once a md file is open, with no need to execute the command manually. Unfortunately, `shell commands` does not provide the most relevant trigger event but only with a `switching the active pane`. So when using built-in `Toggle fold properties in current file`, switch back to a tab with properties already folded would expand them unexpectedly.

**Reply by chrisgurney** [reply]
@XMB-7 Interesting. Just to be clear, you're talking about [this plugin](https://github.com/Taitava/obsidian-shellcommands), correct? I've just been using Shell Commands to execute Unix commands and Python scripts. What does your command look like?

It sounds like [Advanced URI](https://github.com/Vinzent03/obsidian-advanced-uri) is more appropriate in general -- looks like it lets you craft a URI to open a file then execute a command.

So back to what's needed: Is it that you want to open a file, and make sure the properties are folded (but not hidden) on open? Is that what you're looking for? (Just so I'm 100% clear.)

The other command in my plugin --`Show Properties` -- handles expanding folded properties (not just if they're hidden).

**Reply by XMB-7** [reply]
> So is it that you want to open a file, and make sure the properties are folded (but not hidden) on open? 

Yes, that is what I'm looking for. I can achieve this via Shell Commands by combing several command URIs from several third-party plugins: moving cursor to the top, fold more, navigate back the cursor position. But sometimes the cursor position fails to restore, so I've been looking for a single command for folding but not toggling properties.

---

Shell Commands can be triggered automatically on various kinds of events. The URI below belongs to the built-in command `Toggle fold properties in current file`. It can automatically toggle properties every time you switch to a tab (changing focus).

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/65625632/8f8597ae-315b-49d4-9786-29ea3772a4fa)

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/65625632/9c088f90-5e91-49ca-b69c-fc3c9fff9918)

**Reply by chrisgurney** [reply]
OK, thanks for explaining this!

I don't mind adding this in the next release. It will use the same technique as I use in `Show Properties`, which is to find the metadata heading element, and issue a click on the heading. The new `Fold Properties` command will just issue the click only if it's not already collapsed. As for when this will be available, I'm still ironing out the [1.8 release](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/86) but this command can be part of this, as it's minimal work with no expected side-effects / regressions. 

FYI as an aside in case it helps, I don't think for your case you need to execute Shell Commands from Note Toolbar: Just [copy the URI](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/How-to-get-a-URI-with-Advanced-URI) from Advanced URI directly into a toolbar item (like in your Shell Command example above), and it will work the same... as well as work on a phone (as your approach requires the `start` command).

**Reply by XMB-7** [reply]
Thank you for considering adding such a command. 
But without `Shell Command` the command URI cannot be executed automatically, right? Even if I add it as a toolbar item, I still have to manually click on the button to trigger it every time I open a new md file, which is quite overwhelming.

**Reply by chrisgurney** [reply]
@XMB-7 I guess I don't fully understand your workflow and how you're executing those Shell Commands. 😅

Maybe give it a try just putting the URI in a toolbar item and see if it also executes the command. I don't see why it shouldn't.

Whatever works for you, I guess! 😊

**Reply by XMB-7** [reply]
When using Shell Commands to execute the uri of command `Toggle fold properties in current file`: 
![recording](https://github.com/chrisgurney/obsidian-note-toolbar/assets/65625632/e5a3b017-e64d-4046-8227-12973dc7b483)


When using toolbar button to execute the uri of command `Toggle fold properties in current file`: 
![recording](https://github.com/chrisgurney/obsidian-note-toolbar/assets/65625632/1149d080-8ebb-4f58-b65a-d37debeee253)


It seems that no other plugins could also provide hooks like Shell Commands to execute a command on detected events. That's how Shell Commands is necessary in my workflow.

**Reply by chrisgurney** [reply]
I think I got it. So it sounds like once the command's been added to Note Toolbar, you'll be able to just use Advanced URI, and shouldn't need Shell Commands, at least for that use case. Correct?

**Reply by XMB-7** [reply]
Actually just the opposite. To run a command in the current file once switching from one tab to another, I can choose to only use the core plugin Command Palette, or Advanced URI, or Note Toolbar, or other plugins. However, they all require manual triggering by typing in the command name, clicking the [<link name>](<command URI>) link, or clicking on the corresponding toolbar button like above gif shows.
So for automatically executing a command in the current file once switching from one tab to another, Shell Commands comes to rescue. It is indispensable, and I could only use Shell Commands to achieve the goal, if provided the target command URI. All I need to do is to switch tabs, and Shell Commands will automatically detect this tab switching behavior then help trigger the command to fold/unfold the properties.

**Reply by chrisgurney** [reply]
I'm sorry I still don't quite understand -- I'll have to re-read everything when I'm thinking straight. Anyway, glad I can help by adding that command. 😊

**Reply by XMB-7** [reply]
Maybe my fault not explaining it well, since English is not my native language. But in short is that, in the first gif I don't need to do anything to run the target command, while in the second gif I have to click the Fold Properties button to run the target command.
Anyway, thanks again for your time and effort. This plugin also helps a lot in other use cases.

**Reply by chrisgurney** [reply]
@XMB-7 I gave this another read this morning, and it makes more sense to me now!

**Comment by XMB-7** [comment]
Hi, I just tested the added command, and it works pretty well. Thank you. However, besides `toggle properties` and `fold properties` commands, could you also please add another `unfold properties` command that explicitly expands the properties? It should work just in the opposite way to the "fold properties" command, i.e. always keeping the properties unfolded.

**Reply by chrisgurney** [reply]
Look for the [command](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Commands) `Note Toolbar: Show Properties`, which should do what you want. 🙂

Let me know how it works out for you!

**Reply by XMB-7** [reply]
Yeah, it works as I expected. Thanks.
